src/public/components/SignX/SignX.js

```diff
@@ -0,0 +1,146 @@
+import AjaxModule from '../../modules/ajax.js';
+import ButtonComponent from '../Button/Button.js';
+import InputComponent from '../Input/Input.js';
+
+const noop = () => null;
+
+export default class SignXComponent {
+    constructor ({
+        parent = document.body,
+        isSignup = false,
+        afterSuccessSubmit = noop
+    } = {}) {
+        this._parent = parent;
+        this._isSignup = isSignup;
+        this._afterSuccessSubmit = afterSuccessSubmit;
+        this._path = isSignup ? '/signup' : '/signin';
+    }
+
+    _addFormError (error) {
+        this._errorDiv.textContent = error;
+        this._errorDiv.dataset.section = 'error';
+        // this._errorDiv.style.visibility = 'visible';
+        this._errorDiv.display = 'block';
+    }
+
+    _removeFormError () {
+        this._errorDiv.textContent = '';
+        this._errorDiv.display = 'none';
+        console.log(this._errorDiv);
+    }
+
+    _onSubmit (xhr) {
+        if (xhr.status === 400) {
+            const response = JSON.parse(xhr.responseText);
+            console.log(this);
+            this._addFormError(response.error);
+            return;
+        }
+
+        alert('Hello from /SignX: ' + xhr.responseText);
+
+        this._afterSuccessSubmit();
+    }
+
+    _onFocus () {
+        if (this._errorDiv.display === 'block') {
+            console.log(this._errorDiv);
+            this._removeFormError();
+        }
+    }
+
+    _onDestroy () {
+        this._form.removeEventListener('click', );
+    }
+
+    render () {
+        const name = this._isSignup ? 'signup' : 'signin';
+
+        const signXButton = new ButtonComponent({ name,
+            classes: [
+                'sign-x-form__button_width_m',
+                'sign-x-form__button_height_h'] });
```

Comment:
Обычно скобки переносят на следующую строку:
```suggestion
                'sign-x-form__button_height_h'
            ] });
```